### DISCUSSION TITLE
Extract Method Refactoring Granular Detection

### DISCUSSION THREAD

**Discussion by IP1102** [tsantalis/RefactoringMiner#659] [Q&A] [answered]
Hello @tsantalis, I wanted to ask if there's a way to detect whether _extract method refactoring_ was performed, given the un-refactored `class` and the refactored `class` contents? I mean is there any such API exposed or a shortcut to achieve this?

**Comment by tsantalis** [comment, marked as answer]
Hi @IP1102 
You can either use the API that takes a pair of files or directories as input
https://github.com/tsantalis/RefactoringMiner?tab=readme-ov-file#with-two-directories-containing-java-source-code

or you can use the API that takes raw string file contents as input
https://github.com/tsantalis/RefactoringMiner?tab=readme-ov-file#with-file-contents-as-strings

**Reply by IP1102** [reply]
Hello @tsantalis , thank you for the response. This is really helpful!
